src/Log/README.md

```diff
@@ -0,0 +1,116 @@
+# Log
+
+The Log package is a [psr/log](https://github.com/php-fig/log) compatible logger implementation using WordPress actions to log messages using [Monolog](https://github.com/Seldaek/monolog) behind the scenes.
+
+As this uses WordPress actions, it doesn't couple any logger to your code, and simply won't fire any log
+messages if no logger is configured.
+
+## Quick Start
+
+Register `Log_Definer.php` and `Log_Subscriber.php` in Square One Core.php. Call the Log Actions below
+in your code.
+
+## Log Action Definition
+
+```php
+/**
+ * @param string $logger_hook Use on of the constants defined in Log_Actions.
+ * @param string $message The message to log.
+ * @param mixed[] $additional_data_to_log Additional data to log. 
+ */
+do_action( string $logger_hook, string $message, array $additional_data_to_log );
+```
+
+## Example Usage
+
+```php
+do_action( \Tribe\Libs\Log\Log_Actions::INFO, 'Starting connection...' );
+
+// ... more logic
+
+if ( $status !== 200 ) {
+    do_action( \Tribe\Libs\Log\Log_Actions::ERROR, 'Invalid status returned from API.', [ 'status' => $status ] );
+}
+
+```
+## Log Action List
+
+```php
+do_action( Log_Actions::EMERGENCY, 'This is an emergency!', [ 'emergency' ] );
+do_action( Log_Actions::ALERT, 'This is an alert.', [ 'alert' ] );
+do_action( Log_Actions::CRITICAL, 'This is a critical!', [ 'critical' ] );
+do_action( Log_Actions::ERROR, 'This is an error.', [ 'error' ] );
+do_action( Log_Actions::WARNING, 'This is a warning.', [ 'warning' ] );
+do_action( Log_Actions::NOTICE, 'This is a notice.', [ 'notice' ] );
+do_action( Log_Actions::INFO, 'This is an info message.', [ 'info' ] );
+do_action( Log_Actions::DEBUG, 'This is a debug message.', [ 'debug' ] );
+```
+
+## Configuration
+
+Logs are written to the `/wp-content` folder, in a file called `square-one-<today's date>.log`. 
+
+> **NOTE** be cautious of what is being written to log files and where they are stored to ensure no sensitive information is leaked if used on a production server.
+
+**Change Log Path and Name**
+
+```php
+add_filter( 'tribe/log/path', static fn ( $path ) => '/server/path/to/my.log' );
+```
+
+**Change the Default Log Level**
+
+The default level is `DEBUG`, you can pass a [LogLevel Constant](https://github.com/php-fig/log/blob/master/src/LogLevel.php) in a define to configure a different level.
+
+in wp-config.php
+
+```php
+define( 'TRIBE_LOG_LEVEL', \Psr\Log\LogLevel::EMERGENCY );
```

Comment:
It's not clear to me the Level selected, what messages will I see? Are they in order from above? For example, if I set the level to 
`define( 'TRIBE_LOG_LEVEL', \Psr\Log\LogLevel::INFO );` do I see everything above the INFO level? Or only the INFO level? What about DEBUG, is that show all errors? I think this is implied but not clear.

Reply:
This is actually coupled with the `bool $bubble` parameters. So it really depends on how many Loggers you have in the stack and which position in the stack they are in. 

This seems like a good explanation: https://logtail.com/tutorials/how-to-start-logging-with-monolog/#step-6-%E2%80%94-multiple-handlers

Reply:
To properly answer your question, yes you can configure the Handler stack to have certain logs go a certain place, or one log message go to 30 different places at once. Just depends on the developer configures the stack in the Definer.

A great use case is displaying an error/info or what not to a user using a WP-CLI command, but at the same time logging those exact messages to a file as well with an additional Logger, making normally transient information more permanent.